File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    var answer = 0;
    let wlist = []
    let hlist = [];
    sizes.forEach((ele) => {
        let [w, h] = ele;
        if (w > h) {
            wlist.push(w);
            hlist.push(h);
        }
        else {
            wlist.push(h);
            hlist.push(w);
        }
    })
    answer = Math.max(...wlist) * Math.max(...hlist);
    
    return answer;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    priority_queue<int> w;
    priority_queue<int> h;
    for (auto card : sizes) {
        int a = card[0];
        int b = card[1];
        if (a > b) {
            w.push(a);
            h.push(b);
        } else {
            w.push(b);
            h.push(a);
        }
    }
    answer = w.top() * h.top();
    return answer;
}